docs/BBA-EN.md

```diff
@@ -0,0 +1,189 @@
+# Binary Byzantine Agreement
+
+## Overview
+
+Binary Byzantine Agreement tries to solve consensus problem in asynchronous distributed systems where processes or node can commit Byzantine failures. A process can have a Byzantine behavior when it arbitrary deviates from its intended behavior, it then commits a Byzantine failure.
+
+This bad behavior can be intentional or simply the result of temporary fault that altered the its normal behavior in unpredictable way. One example of bad behavior is crash of process.
+
+
+
+## Binary Byzantine consensus problem
+
+The Binary Byzantine consensus problem is that each correct process `p(i)` proposes a value `v(i)` (0 or 1). And each of members in network has to decide a value such that the following properties are satisfied:
+
+* A decided value was proposed by a correct process
+* No two correct processes decide different values
+* A correct process decides at most once.
+* Each correct process decides, and terminate consensus
+
+These properties states that value proposed only by faulty node can be decided. In other words, if all correct processes propose the same value `v`, the value `v'` cannot be decided. 
+
+
+
+## Requirements
+
+The requirements for this algorithm to work is if `n` is the number of members that engaged in the agreement, **the number of failed member `t` must satisfies `n > 3t` condition.**
+
+
+
+## Binary-Value Broadcast
+
+In a BV-broadcast, each correct process `p(i)` BV-broadcasts a binary value and obtains binary values. To store the values obtained by other processes, BV-broadcast provides `binValues(i)` , which initialized to zero-set, which increases when new values are received. And BV-broadcast follows four properties:
+
+* If at least `t + 1` correct processes BV-broadcast the same value `v`, `v` is added to the set `binValues(i)` of each correct process `p(i)`
+* If `p(i)` is correct and `v` is contained in `binValues(i)` , then `v` has been BV-broadcast by a correct process.
+* If a value `v` is added to the set `binValues(i)` , of correct process `p(i)`, eventually `v` which is contained in `binValues(i)` is added to `binValues(j)` at every correct process `p(j)`
+* Eventually the set `binValues(i)` of each correct process `p(i)` is not empty
+
+```go
+func BVBroadcast(v Value) {
+    broadcast(BVal(v))
+    go func() {
+        for {
+            select {
+            case bVal := <- bValChan:
```

Comment:
What is meaning of this statement?
bVal is assigned as bValChan?

Reply:
this is pseudo code, and I described receiving other node's BVAL, as receiving value from channel.

Meaning:

everytime other node send us (local node) BVAL, we should listen and receive those value and handle them. And I described receiving BVAL as receiving value from channel